tests/Functional/Platform/AlterUuidColumnTest.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Functional\Platform;
+
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+use Doctrine\DBAL\Types\GuidType;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\DBAL\Types\Types;
+
+use function sprintf;
+
+class AlterUuidColumnTest extends FunctionalTestCase
```

Comment:
I think you want to limit this test to MariaDB 10.7 and higher because we're specifically testing the migration from old table definitions on that platform.